coverage: poll: Add missing poll event to improve code coverage

pswarnak opened the pull request:
Add testcase for validating K_POLL_TYPE_IGNORE poll event to
improve code coverage. Also, remove few redundant code in kernel.

Signed-off-by: Praful Swarnakar <praful.swarnakar@intel.com>

andyross commented on kernel/poll.c:
```diff
@@ -67,7 +67,6 @@ static inline int is_condition_met(struct k_poll_event *event, u32_t *state)
 		return 0;
 	default:
 		__ASSERT(0, "invalid event type (0x%x)\n", event->type);
-		break;
```
So obviously that was a noop.  I don't disagree with removing it, but can you put unrelated cleanup in its own patch?  Having unrelated junk sitting around makes tasks like test reversions and bisections more likely to fail spuriously.  The commit message says this is a change to "tests", it shouldn't be hammering kernel files.

pswarnak replied:
Thanks for the input. I won't be able to change the code for next two days due to power shutdown issue here. I will remove the kernel changes from this commit and create a separate commit only with kernel code changes as soon i get access to my Linux machine.

pswarnak replied:
Updated as per suggestion. Created two different commits with kernel and test code separately. Kindly check. Thanks again.

pswarnak replied:
@andyross:
Kindly review the changes.

andyross reviewed (dismissed):
Please split out the changes to kernel.  In general cleanup and refactoring should never live in the same patch as functional changes.  Squeezes just barely across my threshold for a -1.   No other complaints.

pswarnak commented:
shippable is failing on mimxrt1050_evk of following error which is not related to the current changes:

/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h: In function ‘LPUART_WriteByte’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/ext/hal/nxp/mcux/drivers/imx/./fsl_lpuart.h:602:9: error: ‘LPUART_Type {aka struct <anonymous>}’ has no member named ‘data’
     base->DATA = data;
         ^~

pswarnak committed:
coverage: kernel: poll: Cleanup redundant code to improve coverage

Remove few redundant code in kernel polling interface.

Signed-off-by: Praful Swarnakar <praful.swarnakar@intel.com>

pswarnak commented:
Shippable failure is already tracked in #9054. It is not related to the current code changes. Kindly review accordingly.

pswarnak committed:
coverage: tests: poll: Add missing poll event to improve code coverage

Add testcase for validating K_POLL_TYPE_IGNORE poll event to
improve code coverage.

Signed-off-by: Praful Swarnakar <praful.swarnakar@intel.com>

nashif commented:
recheck

pswarnak commented:
Kindly let me know, if any action item missing from my end to get this code Merged.
